src/processors/default/ipfsMediaUploader.ts

```diff
@@ -0,0 +1,77 @@
+
+import { urlSource } from 'ipfs-http-client';
+import _ from 'lodash';
+
+
+import { Table } from '../../db/db';
+import { artworkNotOnIpfs, audioNotOnIpfs } from '../../triggers/ipfs';
+import { IPFSFile } from '../../types/ipfsFIle';
+import { Clients, Processor } from '../../types/processor';
+import { ProcessedTrack } from '../../types/track';
+import { rollPromises } from '../../utils/rollingPromises';
+
+function processorFunction(sourceField: 'lossyAudioURL' | 'lossyArtworkURL', replaceField: 'lossyAudioIPFSHash' | 'lossyArtworkIPFSHash') {
+  return async (tracks: ProcessedTrack[], clients: Clients) => {
+    const updates: ProcessedTrack[] = [];
+    const ipfsFiles: IPFSFile[] = [];
+
+    const existingFiles = await clients.db.getRecords<IPFSFile>(Table.ipfsFiles, [
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      ['whereIn', ['url', tracks.map(track => track[sourceField])]]
+    ]);
+    const fileByUrl = _.keyBy(existingFiles, 'url')
```

Comment:
you're already joining in the trigger with files, so the url/cid mapping should already be available without having to do this extra db query right?

because the processor is actually getting an array of tracksWithFiles, not just tracks ?

Reply:
At the moment the processor does just get the track data. I added this extra query to see if any of the files for those tracks have already been uploaded. I originally designed the trigger and the processor to just work with tracks but I'll include the data from the join so we can remove the extra query